test/libsolidity/semanticTests/functionCall/failed_create.sol

```diff
@@ -14,15 +14,25 @@ contract C {
 	}
 }
 // ====
-// compileViaYul: also
 // EVMVersion: >=byzantium
+// compileViaYul: also
 // ----
 // constructor(), 20 wei
+// gas ir: 437069
+// gas legacy: 324594
+// gas legacyOptimized: 222226
+// gas irOptimized: 232551
 // f(uint256): 20 -> 1370859564726510389319704988634906228201275401179
 // x() -> 1
 // f(uint256): 20 -> FAILURE
 // x() -> 1
 // stack(uint256): 1023 -> FAILURE
+// gas ir: 1162959
+// gas legacy: 981671
+// gas legacyOptimized: 831459
+// gas irOptimized: 835314
 // x() -> 1
 // stack(uint256): 10 -> 693016686122178122849713379390321835634789309880
+// gas ir: 106194
```

Comment:
What exactly decides when the gas measurement for optimized code need to be done?